third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -124,30 +124,35 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
   datas.reserve(tensors.size());
   int64_t total_size = 0;
   for (auto& tensor : tensors) {
-    PjRtDevice* pjrt_device = StringToPjRtDevice(tensor.device);
-
-    auto literal = std::make_shared<xla::Literal>(tensor.shape);
-    tensor.populate_fn(tensor, literal->untyped_data(), literal->size_bytes());
-    std::vector<int64_t> byte_strides(literal->shape().dimensions_size());
-    XLA_CHECK_OK(
-        ShapeUtil::ByteStrides(literal->shape(), absl::MakeSpan(byte_strides)));
-    total_size += literal->size_bytes();
+    ComputationClient::DataPtr data;
+    if (tensor.device == "SPMD:0") {
```

Comment:
Do we need the placeholder data for the current flag interface as well? In any case, let's clarify + comment some more about this route.

Reply:
Not sure if this is straightforward to do, but it could be cleaner if we can do the check and call or not call the TransferToServer --then we don't need to change the runtime client APIs.

Reply:
Done - added a comment and moved this up to tensor_util.cpp. We still need to use the placeholder data, even with the flag, but we can set the "SPMD:0" device internally based on the flag so that it's 100% transparent to the user.